### PR TITLE
Infiinte Loop with MaxDate for GetOccurrences

### PR DESCRIPTION
**Pull request by dsbegnoche**
Fixes bug when GetOccurrences ends on MaxDate and only Count is set, an
error is thrown inside an infinite loop

### CODE REVIEW

File: net-core/Ical.Net/Evaluation/RecurrencePatternEvaluator.cs [hunk 1]
```diff
@@ -259,11 +259,21 @@ private void EnforceEvaluationRestrictions(RecurrencePattern pattern)
                     break;
                 }
 
+                if (pattern.Until != DateTime.MinValue && candidate != DateTime.MinValue && candidate > pattern.Until)
```
Comment (pomoika) [inline net-core/Ical.Net/Evaluation/RecurrencePatternEvaluator.cs]: This condition is exact duplicate of the above statement, therefore it should be removed.

Comment (dsbegnoche) [reply net-core/Ical.Net/Evaluation/RecurrencePatternEvaluator.cs]: Good catch.  This was an accident.  Removed the duplication.  @pomoika

**Comment by mscrivo** [comment]
Running into this as well after upgrading from 3.0.6 to 4.1.2 .. any reason this hasn't been merged in yet?

**Comment by dsbegnoche** [comment]
@rianjs Any chance we can get this reviewed and merged?  This bug is preventing us from upgrading to latest version.

**Comment by rianjs** [comment]
https://www.nuget.org/packages/Ical.Net/4.2.0

**Pull request closed without merging**